extension-style-app/src/androidTest/java/com/mapbox/maps/testapp/style/sources/generated/GeoJsonSourceTest.kt

```diff
@@ -450,10 +549,26 @@ class GeoJsonSourceTest : BaseStyleTest() {
     val testSource = geoJsonSource("testId") {
       url(TEST_URI)
     }
-    setupSource(testSource)
-    testSource.geometry(feature.geometry()!!)
-    // Plain json string data getter is not supported due to performance consideration.
-    Handler(Looper.getMainLooper()).postDelayed({ assertEquals(null, testSource.data) }, 10L)
+    val listener = OnStyleDataLoadedListener {
+      if (it.type == StyleDataType.SOURCES) {
```

Comment:
AFAIR for better readability let's check for actual `testId` here. `SOURCES` works in that test as we have one source in total but for general case and perhaps as a possible reference let's do it fully correct way.

Reply:
`testId` can be verified with `OnSourceDataLoadedListener`. my initial idea was to check with `OnSourceDataLoadedListener`, which actually make sense to test. but I saw in the ticket and in the geojson source file that we need to check `OnStyleDataLoadedListener` in order to verify the data is loaded.